### PR TITLE
CDRIVER-4085 fix docs for server_connection_id getters

### PR DESCRIPTION
**Pull request by jmikola**
This removes copypasta that was missed in https://github.com/mongodb/mongo-c-driver/commit/ba73833f18191b34acc990bdca91bb9e68c90dfb (#972)

### CODE REVIEW

File: src/libmongoc/doc/mongoc_apm_command_failed_get_server_connection_id.rst [hunk 1]
```diff
@@ -22,7 +27,7 @@ Parameters
 Returns
 -------
 
-The pointer passed with :symbol:`mongoc_client_set_apm_callbacks` or :symbol:`mongoc_client_pool_set_apm_callbacks`.
+The server connection ID as a positive integer or -1 if it is not available.
```
Comment (jmikola) [inline src/libmongoc/doc/mongoc_apm_command_failed_get_server_connection_id.rst:30]: https://github.com/mongodb/mongo-c-driver/commit/ba73833f18191b34acc990bdca91bb9e68c90dfb#diff-de5aea95eea3d2b786d33b8b694d349c445847e9cc506502c6824e05f177cc0cR48 defined `MONGOC_NO_SERVER_CONNECTION_ID` internally as `-1`, which is similar to what was done for `MONGOC_RTT_UNSET`.

I previously opened a ticket to document the unset RTT return value in [CDRIVER-4176](https://jira.mongodb.org/browse/CDRIVER-4176) and @kevinAlbs proposed just making `MONGOC_RTT_UNSET` public. Should we just consider doing so for `MONGOC_NO_SERVER_CONNECTION_ID`?

If so, would we want to keep the same name or change it to `MONGOC_SERVER_CONNECTION_ID_UNSET`?

Comment (rcsanchez97) [reply src/libmongoc/doc/mongoc_apm_command_failed_get_server_connection_id.rst:30]: So, I initially considered using `MONGOC_SERVER_CONNECTION_ID_UNSET`.  However, it seemed to me like "unset" was something different from "not present".  Based on the naming of other nearby constants, like `MONGOC_NO_SESSIONS` and `MONGOC_NO_SET_VERSION`, the better name appeared to be `MONGOC_NO_SERVER_CONNECTION_ID`.  That said, I'm not opposed to your suggested change.  I simply wanted to state my rationale more thoroughly for your consideration.

**Review by rcsanchez97 — APPROVED** [review]
LGTM

Thanks for catching these oversights.  I was so focused on making sure that I documented each function and that I named the files correctly, that I completely forgot to read the documentation after copying it to make any necessary changes.  :facepalm:

File: src/libmongoc/doc/mongoc_apm_command_failed_get_server_connection_id.rst (hunk 1 above)
Comment (eramongodb) [reply src/libmongoc/doc/mongoc_apm_command_failed_get_server_connection_id.rst:30]: I am slightly in favor of `MONGOC_NO_SERVER_CONNECTION_ID` but am not opposed to the `UNSET` naming scheme.

**Review by eramongodb — APPROVED** [review]

File: src/libmongoc/doc/mongoc_apm_command_failed_get_server_connection_id.rst (hunk 1 above)
Comment (jmikola) [reply src/libmongoc/doc/mongoc_apm_command_failed_get_server_connection_id.rst:30]: Ah, that makes much more sense and I agree `MONGOC_NO_SERVER_CONNECTION_ID` is the better choice. I agree that at the time we call this method, we either know that their's an ID or not, which is a different situation than RTT.

Any thoughts on whether we should make the constant public? I'm happy to leave this as-is and just add it as a todo item in [CDRIVER-4176](https://jira.mongodb.org/browse/CDRIVER-4176).

Comment (rcsanchez97) [reply src/libmongoc/doc/mongoc_apm_command_failed_get_server_connection_id.rst:30]: I would lean toward adding it to CDRIVER-4176 and changing the visibility of all of them at the same time.

**Pull request merged**